### PR TITLE
Added BCR-0004 and BCR-0005

### PR DESCRIPTION
**Pull request by wolfmcnally**
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA512

Added BCR-0004 "The BC32 Data Encoding Format" and BCR-0005 "Uniform Resources (UR): Encoding Structured Binary Data for Transport in URIs and QR Codes".	3d03ca2	Wolf McNally <wolf@wolfmcnally.com>	May 5, 2020 at 2:39 AM
-----BEGIN PGP SIGNATURE-----
Comment: GPGTools - https://gpgtools.org

iQIzBAEBCgAdFiEElDZS7jhEF2DD3DU2S2wvz4lHgK4FAl6xNIYACgkQS2wvz4lH
gK5jdw/9GIijuvpi4z44kqSufGy2vVvEtbs64+LEMc3CWASLiTfdP1D9RYORMhUy
HWbpbGduDQCUTz80/4VYSPL9pWb0TsmGikDgNWfIWiG7XC/D6md2cFTo/uq+LMiQ
ax3NM7Je/m45TrSF7k1WRCsy9r9GA8UQ/NcOdQUCCeLsz7LrR1e7cwwiSzIPaGoO
+QsXOrqrSjm+eBi+BoHkPauAHXIbQ7M8rc9ksOJCAOvw/+O6VRZPcpCRPexVKmqU
qmjmEAkZ3iqPt1nqMIPtKLHDNBhEeEPCv/8PZkYpHmxk/nsAlwrhc/7bKxnLtvkx
u7hNm5HTkDR+eL3Ctw9SaVhVwylcNq8Icq4MqbZ3rs/248zUdXmAuNYYQ9EFRoRj
8PcsZ/Emq2BeaKuTguepwYuXCANVRVik2iVu8sG/vncRMYlPINIeRytPG/4R/zoT
AhFT++elhB+ma/U0UaV1AOvz1bA1n8dBLPjwXTBYjFrFuAhQHdBqhv8K4WAIXDxL
Bat39RA6ck69vjX25frYiUtdOwJKdkUHOq7ZTmEwaRd6WFrgUDvJvvHYFMvJ+hQP
/nmaebKkHBHtXk9gWFFPDYpXOJ/AsF40bhOHgzK/iatbKJcmvmj3BtxwludQIDMa
b+BPJCENa5DAZuEIm0eCuBPeLfrL2Uhi6F+pLWZ9alY8I1YvfnM=
=3Mmx
-----END PGP SIGNATURE-----

### CODE REVIEW

File: papers/bcr-0004-bc32.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+Author: Wolf McNally<br/>
+Date: May 2, 2020
+
+---
+
+### Introduction
+
+The [Bech32] checksummed base32 format was introduced to transport segregated witness signatures for Bitcoin. It has several desirable qualities:
+
+* The payload and checksum data use a limited character set compatible with both URI syntax and QR code alphanumeric encoding mode.
```
Comment (ChristopherA) [inline papers/bcr-0004-bc32.md:15]: encoding mode with its built-in compression.

File: papers/bcr-0004-bc32.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+* The character set uses the letters A-Z in a case-modal fashion (i.e., one can use all upper case or all lower case but the algorithm rejects mixed-case.)
+* The character set is chosen to minimize ambiguity, and the ordering is chosen to minimize the number of pairs of similar characters.
+* Built-in error detection.
```
Comment (ChristopherA) [inline papers/bcr-0004-bc32.md]: …error detection, and option error identification for smaller <80 character payloads.

File: papers/bcr-0004-bc32.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+* Because the character set used are all "word characters", most text editors will select the entire sequence of characters when it is double-clicked.
+
+However, it also has a couple drawbacks when considered for general data encoding:
+
+* The checksum algorithm is suited to a smaller number of bytes.
```
Comment (ChristopherA) [inline papers/bcr-0004-bc32.md]: Peter says it is only the error correction that is limited in bytes

File: papers/bcr-0004-bc32.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+* The characters allowed for the human readable part ("HRP") conflict with both URI syntax and QR code alphanumeric encoding mode.
+
+### Proposal
+
+This document proposes a new, more general data encoding format *BC32* that is based on Bech32 but makes the following modifications:
+
+* The HRP and numeral `1` divider are no longer included.
+* **TODO:** The checksum is enhanced to handle larger payloads.
```
Comment (ChristopherA) [inline papers/bcr-0004-bc32.md]: parameters are defined to handle error correction / identification for larger payloads.

File: papers/bcr-0004-bc32.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
+# The BC32 Data Encoding Format
+## BCR-0004
+
+**© 2020 Blockchain Commons**
+
+Author: Wolf McNally<br/>
```
Comment (ChristopherA) [inline papers/bcr-0004-bc32.md]: You certainly are the lead author on this, but I've been driving this agenda and will be driving it through standards, so can you add my name last?

File: papers/bcr-0004-bc32.md (hunk 1 above)
Comment (wolfmcnally) [reply papers/bcr-0004-bc32.md:15]: It's not really "compression", it's merely higher efficiency than encoding generic ASCII or base64. I'll add a note to this effect.

File: papers/bcr-0004-bc32.md (hunk 2 above)
Comment (wolfmcnally) [reply papers/bcr-0004-bc32.md]: Fixed.

File: papers/bcr-0004-bc32.md (hunk 3 above)
Comment (wolfmcnally) [reply papers/bcr-0004-bc32.md]: Fixed. Added a note about HRP not being useful for general structured data type identification.

File: papers/bcr-0004-bc32.md (hunk 4 above)
Comment (wolfmcnally) [reply papers/bcr-0004-bc32.md]: Fixed.

File: papers/bcr-0004-bc32.md (hunk 5 above)
Comment (wolfmcnally) [reply papers/bcr-0004-bc32.md]: Absolutely. Added your name to this and BCR-0005.

**Review by ChristopherA — APPROVED** [review]
+1 LGTM. Thanks!

**Pull request merged**